### PR TITLE
Add `Seek` trait

### PR DESCRIPTION
**Pull request by smindinvern**
I noticed this was missing and would really benefit my use case.  I'll be happy to address any feedback as best I can to get this merged.

### CODE REVIEW

File: src/adapters/tokio.rs [hunk 1]
```diff
@@ -72,6 +72,29 @@ impl<T: tokio::io::AsyncWrite + Unpin + ?Sized> crate::asynch::Write for FromTok
...
+    where
+        Self: 'a;
+
+    fn seek<'a>(&'a mut self, pos: crate::SeekFrom) -> Self::SeekFuture<'a> {
+        poll_fn::poll_fn(move |cx| {
+            // Note: `start_seek` can return an error if there is another seek in progress.
+            // Therefor it is recommended to call `poll_complete` before any call to `start_seek`.
+            match Pin::new(&mut self.inner).poll_complete(cx) {
+                Poll::Ready(r) => match r {
+                    Ok(_) => match Pin::new(&mut self.inner).start_seek(pos.into()) {
```
Comment (Dirbaio) [inline src/adapters/tokio.rs:86]: I don't think the logic here is correct.  You `poll_complete`, then `start_seek`, then `poll_complete`, which can return Pending. If it does, `poll_fn` will call the closure again when woken, which will `poll_complete`, and then `start_seek` **again**.

File: src/blocking.rs [hunk 2]
```diff
@@ -133,6 +133,25 @@ pub trait Write: crate::Io {
...
+/// Blocking seek within streams.
+///
+/// Semantics are the same as [`std::io::Seek`], check its documentation for details.
+pub trait Seek: crate::Io {
+    /// Seek to an offset, in bytes, in a stream.
+    fn seek(&mut self, pos: crate::SeekFrom) -> Result<u64, Self::Error>;
+
+    /// Rewind to the beginning of a stream.
+    fn rewind(&mut self) -> Result<(), Self::Error> {
+        self.seek(crate::SeekFrom::Start(0))?;
```
Comment (Dirbaio) [inline src/blocking.rs:145]: you can remove the `?` and the `Ok(())`

File: src/lib.rs [hunk 3]
```diff
@@ -50,6 +50,44 @@ impl Error for ErrorKind {
...
+    Start(u64),
+    /// Sets the offset to the size of this object plus the specified number of bytes.
+    End(i64),
+    /// Sets the offset to the current position plus the specified number of bytes.
+    Current(i64),
+}
+
+#[cfg(feature = "std")]
+#[cfg_attr(docsrs, doc(cfg(feature = "std")))]
+impl Into<std::io::SeekFrom> for SeekFrom {
```
Comment (Dirbaio) [inline src/lib.rs:69]: to implement conversions you have to impl `From`, not `Into` (`Into` is blanket-impl'd on `From`)

File: src/adapters/futures_io.rs [hunk 4]
```diff
@@ -64,6 +64,21 @@ impl<T: futures::io::AsyncWrite + Unpin + ?Sized> crate::asynch::Write for FromF
     }
 }
 
+impl<T: futures::io::AsyncSeek + Unpin + ?Sized> crate::asynch::Seek for FromFutures<T> {
+    type SeekFuture<'a> = impl Future<Output = Result<u64, Self::Error>>
+    where
+        Self: 'a;
+
+    fn seek<'a>(&'a mut self, pos: crate::SeekFrom) -> Self::SeekFuture<'a> {
+        let pos = match pos {
```
Comment (Dirbaio) [inline src/adapters/futures_io.rs]: `futures::io::SeekFrom` is a reexport of `std::io::SeekFrom` so you can use the conversion impls you did, with `.into()`

Comment (Dirbaio) [reply src/adapters/futures_io.rs]: oh you just did, nice!

File: src/lib.rs (hunk 3 above)
Comment (smindinvern) [reply src/lib.rs]: Thanks, mis-remembered that.  Done.

File: src/blocking.rs (hunk 2 above)
Comment (Dirbaio) [reply src/blocking.rs:145]: ah my bad, I assumed `seek` returned `Result<(), Self::Error>` as well. Sorry!

File: src/adapters/tokio.rs (hunk 1 above)
Comment (smindinvern) [reply src/adapters/tokio.rs]: Not sure what I was thinking before, but LMK if this makes more sense now.

**Review by Dirbaio — APPROVED** [review]
Looks great now! Thanks for the PR :)

**Pull request merged**